src/AnchoredOverlay/AnchoredOverlay.tsx

```diff
@@ -30,6 +26,11 @@ export interface AnchoredOverlayProps extends Pick<OverlayProps, 'height' | 'wid
    * A callback which is called whenever the overlay is currently open and a "close gesture" is detected.
    */
   onClose?: (gesture: 'click-outside' | 'escape') => unknown
+
+  /**
+   * A callback that occurs when a user presses a mouse button over an element.
+   */
+  onMouseDown?: (event: React.MouseEvent) => unknown
```

Comment:
I would match the `ActionMenu` and use `onOverlayMouseDown` here since we still technically have a split between anchor and `Overlay` in this component